Update to latest `librustzcash`, `zaino` and `zebrad`

nuttycom opened the pull request:
Updates to:
* `librustzcash 6637660ea340c50d22ea0cf72df8f9452243652c`
* `zaino-* 642261116cbaf0c7a27ac37f58060164cd686575`
* `zebra-* 2.0`

str4d commented on zallet/src/components/json_rpc/methods/list_unspent.rs:
```diff
@@ -84,16 +87,16 @@ pub(crate) struct UnspentNote {
 pub(crate) fn call(wallet: &DbConnection) -> Response {
-    // Use the height of the maximum scanned block as the anchor height, to emulate a
-    // zero-conf transaction in order to select every note in the wallet.
-    let anchor_height = match wallet.block_max_scanned().map_err(|e| {
```
We already have an issue for this: #139

nuttycom commented:
[force-pushed](https://github.com/zcash/wallet/compare/1dddbbc6f9d8158eda97596486cb569092a39a71..828261404a8a498339174b39116118da495633ae) to rebase on `main` and address comments from code review. See https://github.com/zcash/wallet/pull/203/commits/828261404a8a498339174b39116118da495633ae for changes since last review.

nuttycom commented:
Converted to draft pending resolution of https://github.com/ZcashFoundation/zebra/issues/9817

str4d reviewed (approved):
utACK 9f256728beb3bf7dfc7065ca7141a3c763841139 modulo waiting to confirm that https://github.com/ZcashFoundation/zebra/pull/9823/commits/5bcda3cba4a46dcf28c423b060d8f088dbb0161d exists in their repo history after release.

nuttycom commented:
Returned to draft on advice to wait for the `zebra-rpc` dependency to be published instead of depending on a patch revision.

nuttycom commented:
[force-pushed](https://github.com/zcash/wallet/compare/9f256728beb3bf7dfc7065ca7141a3c763841139..6f17330d1417a8ba5c37113373808297165a1826) to update to the released `zebra-rpc` patch version.

str4d reviewed (approved):
utACK 6f17330d1417a8ba5c37113373808297165a1826
